### DISCUSSION TITLE
Allow specifying a default target for debugging tests

### DISCUSSION THREAD

**Discussion by Nerixyz** [microsoft/vscode-cmake-tools#3940] [Ideas]
When debugging tests, having the quick picker open up repeatedly is annoying. I'd like to specify a default target that gets used when debugging tests.
I've implemented an MVP in https://github.com/Nerixyz/vscode-cmake-tools/commit/21392c9844f24725f9d353913645713afb2f5b47. You set `cmake.ctest.testSuiteDelimiter.debugLaunchTarget` to the target name (seems like that's the only decent identifier) and if the target exists, it will be used - otherwise the picker is shown. I noticed
https://github.com/microsoft/vscode-cmake-tools/blob/ed6d9adeb33a8d038c8c55d04d2de7e9deb94add/src/ctest.ts#L933

... but I don't know about the status of that, and this would require choosing the target once for every session. Does my MVP go in the right direction to open a PR? (the wording might be a bit confusing as CMake also has "targets", but that doesn't impact the functionality)

**Comment by gcampbell-msft** [comment]
@Nerixyz The best suggestion that I can give is to create a launch.json so that you don't have to utilize the quick debug configuration that our extension uses, but you can have full customization over any launch.json configuration.

**Reply by Nerixyz** [reply]
I do have a launch.json, but I have multiple configurations in there (one for regular debugging and one for tests at least). With multiple configurations, the popup to select a configuration will always show. I'd want the extension to select the one I intended for testing automatically.

**Reply by gcampbell-msft** [reply]
To make sure I understand, you'd like to be able to set a setting in your VS Code setting to define which launch target you want to select whenever you start to debug? 

This seems a reasonable suggestion to me. I took a quick look at your MVP and if you're interested in creating a PR we'd be happy to take a contribution. There may be some iteration on the PR, but we'd be happy to consider any changes you suggest.

**Reply by Nerixyz** [reply]
>  you'd like to be able to set a setting in your VS Code setting to define which launch target you want to select whenever you start to debug?

Yes. Or rather: I'd like to avoid the popup for the launch configuration selection when repeatedly debugging a test.

**Reply by gcampbell-msft** [reply]
Understood. We'd be happy to take a look at a PR contribution. Additionally, if you want us to track this, it'd be best if you created an issue so that we can prioritize it correctly. Thanks!
